providers/pg_database.rb

```diff
@@ -26,14 +31,22 @@ def createdb_command
   cmd << "--template #{new_resource.template}"
   cmd << "--encoding #{new_resource.encoding}"
   cmd << "--owner #{new_resource.owner}" if new_resource.owner
+  cmd << "#{connection_string}" unless connection_string.empty?
   cmd << new_resource.database
   cmd.join(" ")
 end
 
 def database_exist?
   project_name = node['enterprise']['name']
+  connection_string = []
+  connection_string << "--host #{new_resource.host} " if  new_resource.host
+  connection_string << "--username #{new_resource.username} " if  new_resource.username
+  connection_string << "--password #{new_resource.password} " if  new_resource.password
+  connection_string = connection_string.join(" ")
+
   command = <<-EOM.gsub(/\s+/," ").strip!
     psql --dbname template1
+         #{connection_string unless connection_string.empty?}
```

Comment:
I don't think you need the `unless` here, since doing string interpolation automatically does `.to_s`, and adding an empty string to the string won't matter.